### PR TITLE
Fix a lot of papercuts in `sg install`/`sg setup` after runthrough today

### PR DESCRIPTION
**Pull request by mrnugget**
These are the results of the run-through of `sg install`/`sg setup` today after I did a runthrough of setting up a new dev env on a new machine.

### CODE REVIEW

File: dev/sg/sg_install.go [hunk 1]
```diff
@@ -39,7 +39,10 @@ func installExec(ctx context.Context, args []string) error {
 	case "linux":
 		location = filepath.Join(homeDir, ".local", "bin", "sg")
```
Comment (jeanduplessis) [inline dev/sg/sg_install.go:40]: Any reason we don't make this consistent across linux and macos?

**Review by jeanduplessis — APPROVED** [review]

**Review by david-sandy — APPROVED** [review]
I think there are good changes.

File: dev/sg/sg_install.go (hunk 1 above)
Comment (mrnugget) [reply dev/sg/sg_install.go:40]: On Linux `~/.local/bin` is part of standardised/recommended practices. See [this doc on file hierarchy by freedesktop.org](https://www.freedesktop.org/software/systemd/man/file-hierarchy.html#Home%20Directory).

On macOS this convention doesn't exist

Comment (jeanduplessis) [reply dev/sg/sg_install.go:40]: 👍 TIL

**Pull request merged**